src/base/RegexPattern.cc

```diff
@@ -8,34 +8,28 @@
 
 #include "squid.h"
 #include "base/RegexPattern.h"
-#include "base/TextException.h"
 #include "debug/Stream.h"
 #include "sbuf/Stream.h"
 
 #include <iostream>
-#include <utility>
 
-RegexPattern::RegexPattern(const SBuf &aPattern, const int aFlags):
+RegexPattern::RegexPattern(const SBuf &aPattern, const std::regex::flag_type aFlags):
     pattern(aPattern),
-    flags(aFlags)
+    regex(aPattern.rawContent(), aPattern.length(), aFlags)
 {
-    memset(&regex, 0, sizeof(regex)); // paranoid; POSIX does not require this
-    if (const auto errCode = regcomp(&regex, pattern.c_str(), flags)) {
-        char errBuf[256];
-        // for simplicity, ignore any error message truncation
-        (void)regerror(errCode, &regex, errBuf, sizeof(errBuf));
-        // POSIX examples show no regfree(&regex) after a regcomp() error;
-        // presumably, regcom() frees any allocated memory on failures
-        throw TextException(ToSBuf("POSIX regcomp(3) failure: (", errCode, ") ", errBuf,
-                                   Debug::Extra, "regular expression: ", pattern), Here());
-    }
-
+    assert(aFlags != 0);
```

Comment:
Due to the current messy situation with pattern syntax' we cannot enforce std::regex::basic flag existence. Instead for now require that the caller sets the flags explicitly.